compaction/compaction_manager.cc

```diff
@@ -353,7 +353,7 @@ class compaction_manager::major_compaction_task : public compaction_manager::tas
         // candidates are sstables that aren't being operated on by other compaction types.
         // those are eligible for major compaction.
         auto* t = _compacting_table;
-        sstables::compaction_strategy cs = t->get_compaction_strategy();
```

Comment:
I don't understand what is the connection between idempotency and being able to return something by reference.

Reply:
`table::get_compaction_strategy` returns a reference to its compaction_strategy instance, but it was copied here - causing side effects to be done on the copy rather than on the original.

Looks like unintentionally, this protected against #10989, and it can be done regardless of this series, after e1e4a73793c50c9f9aa0ee0fcf0b484bcca24b2d.

Without this patch twcs get_sstables_for_compaction might be a bit less efficient as it will keep looking for fully expired sstables once `now - _last_expired_check > _options.expired_sstable_check_frequency`.

I'll file an issue about that and submit this patch in that context so we can decide whether we want to backport it there.